src/openassetio-python/private/include/openassetio/private/python/pointers.hpp

```diff
@@ -48,10 +49,19 @@ Ptr createPyRetainingPtr(const py::object& pyInstance,
   // of the original shared_ptr holder stored on the PyObject.
   auto pyInstancePtr =
       std::shared_ptr<py::object>{new py::object{pyInstance}, [](py::object* pyObjectPtr) {
-                                    // Acquire the GIL, in case deleter
-                                    // runs in a non-Python thread.
-                                    py::gil_scoped_acquire gil;
-                                    delete pyObjectPtr;
+                                    if (_Py_IsFinalizing()) {
```

Comment:
[question] Im struggling to find any docs on this method. Is it misleadingly named? In that it will also return true if it is already finalized?

Reply:
It's a bit of a wierd one. The leading underscore and lack of explicit docs makes it sound like something internal that shouldn't be called. However, [other docs](https://docs.python.org/3/c-api/init.html) explicitly say we can use it.

And yeah it's misleadingly named as you say.

Reply:
🤣 solid foundations to the core.